### PR TITLE
patch: Add deploy to balena action

### PR DESCRIPTION
**Pull request by vipulgupta2048**
Signed-off-by: Vipul Gupta (@vipulgupta2048) <vipul@balena.io>

### CODE REVIEW

**Comment by vipulgupta2048** [comment]
@balena-ci rebase

File: .github/workflows/balena-ci.yml [hunk 1]
```diff
@@ -5,19 +5,22 @@ on:
...
-  build:
+  balena-cloud-build:
     runs-on: ubuntu-20.04
     strategy:
+      fail-fast: true
       matrix:
-        fail-fast: true
         fleets: [testbot-rig, testbot-rig-beta, testbot-personal]
     steps:
       - uses: actions/checkout@v2
```
Comment (klutchell) [inline .github/workflows/balena-ci.yml:15]:
```suggestion
      - uses: actions/checkout@v2
        with:
          submodules: true
```
We need the contracts submodule.

File: .github/workflows/balena-ci.yml [hunk 2]
```diff
@@ -5,19 +5,22 @@ on:
...
 
 jobs:
-  build:
+  balena-cloud-build:
     runs-on: ubuntu-20.04
     strategy:
+      fail-fast: true
       matrix:
-        fail-fast: true
         fleets: [testbot-rig, testbot-rig-beta, testbot-personal]
```
Comment (klutchell) [inline .github/workflows/balena-ci.yml:13]:
```suggestion
        fleets: [balena/testbot-rig, balena/testbot-rig-beta, balena/testbot-personal]
```
Use fleet slugs to simplify adding new fleets from other orgs if needed.

File: .github/workflows/balena-ci.yml [hunk 3]
```diff
@@ -5,19 +5,22 @@ on:
...
     steps:
       - uses: actions/checkout@v2
-      - uses: balena-io/balena-ci@v0.3.2
+      - uses: balena-io/deploy-to-balena-action@master
+        id: build
         with:
           balena_token: ${{ secrets.BALENA_TOKEN }}
           github_token: ${{ secrets.GITHUB_TOKEN }}
           fleet: balena/${{ matrix.fleets }}
```
Comment (klutchell) [inline .github/workflows/balena-ci.yml:21]:
```suggestion
          fleet: ${{ matrix.fleets }}
```
See related comment above
> Use fleet slugs to simplify adding new fleets from other orgs if needed.

**Review by klutchell — APPROVED** [review]

**Pull request merged**